packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1403,6 +1403,179 @@ export abstract class TargetObjectKeyFormat {
+ */
+export class ReplicationTimeValue {
+  /**
+   * Fifteen minutes.
+   */
+  public static readonly FIFTEEN_MINUTES = new ReplicationTimeValue(15);
+
+  /**
+   * @param minutes the time in minutes
+   */
+  private constructor(public readonly minutes: number) {};
+}
+
+/**
+ * Specifies which Amazon S3 objects to replicate and where to store the replicas.
+ */
+export interface ReplicationRule {
+  /**
+   * A container for information about the replication destination.
+   *
+   * The destination can be either in the same AWS account or a cross account.
+   */
+  readonly destination: ReplicationDestination;
+
+  /**
+   * Specifying S3 Replication Time Control (S3 RTC),
+   * including whether S3 RTC is enabled and the time when all objects and operations on objects must be replicated.
+   *
+   * @default - S3 Replication Time Control is not enabled
+   */
+  readonly replicationTimeControl?: ReplicationTimeValue;
+
+  /**
+   * A container specifying replication metrics-related settings enabling replication metrics and events.
+   *
+   * @default - Replication metrics are not enabled
+   */
+  readonly metrics?: ReplicationTimeValue;
+
+  /**
+   * The customer managed AWS KMS key stored in AWS Key Management Service (KMS) for the destination bucket.
+   * Amazon S3 uses this key to encrypt replica objects.
+   *
+   * Amazon S3 only supports symmetric encryption KMS keys.
+   *
+   * @see https://docs.aws.amazon.com/kms/latest/developerguide/symmetric-asymmetric.html
+   *
+   * @default - Amazon S3 uses the AWS managed KMS key for encryption
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * The storage class to use when replicating objects, such as S3 Standard or reduced redundancy.
+   *
+   * @default - The storage class of the source object
+   */
+  readonly storageClass?: StorageClass;
+
+  /**
+   * Specifies whether Amazon S3 replicates objects created with server-side encryption using an AWS KMS key stored in AWS Key Management Service.
+   *
+   * @default false
+   */
+  readonly sseKmsEncryptedObjects?: boolean;
+
+  /**
+   * Specifies whether Amazon S3 replicates modifications on replicas.
+   *
+   * @default false
+   */
+  readonly replicaModifications?: boolean;
+
+  /**
+   * The priority indicates which rule has precedence whenever two or more replication rules conflict.
+   *
+   * Amazon S3 will attempt to replicate objects according to all replication rules.
+   * However, if there are two or more rules with the same destination bucket,
+   * then objects will be replicated according to the rule with the highest priority.
+   *
+   * The higher the number, the higher the priority.
+   *
+   * It is essential to specify priority explicitly when the replication configuration has multiple rules.
+   *
+   * @default 0
+   */
+  readonly priority?: number;
+
+  /**
+   * Specifies whether Amazon S3 replicates delete markers.
+   *
+   * @default false
```

Comment:
What's the implication of this being `false` ? That replicated objects in versioned buckets don't get deleted?

Reply:
False means do not replicate delete marker to the destination bucket.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/delete-marker-replication.html

I've updated JSDoc.

```ts
  /**
   * Specifies whether Amazon S3 replicates delete markers.
   *
   * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/delete-marker-replication.html
   *
   * @default - delete markers in source bucket is not replicated to destination bucket
   */
  readonly deleteMarkerReplication?: boolean;
```